[MRG] SimpleImputer skipped feature warning with feature names (#21617)
scikit-learn/scikit-learn: doc/whats_new/v1.1.rst

  :pr:`21448` by :user:`Oleh Kozynets <OlehKSS>` and
  :user:`Christian Ritter <chritter>`.

> - |Enhancement| Added warning with feature names for skipped features
>   due to no observed values in :class:`SimpleImputer`.
>   :pr:`21617` by :user:`Christian Ritter <chritter>`.

Reviewer: Might this phrasing be a little bit more direct?
```suggestion
- |Enhancement| :class:`SimpleImputer` now warns with feature names
  when features which are skipped due to unobserved values.
  :pr:`21617` by :user:`Christian Ritter <chritter>`.
```

Reviewer 2: Or even more explicit:

when features which are skipped due to the lack of any observed values in the training set.

Reviewer: Based on @ogrisel's proposal:
```suggestion
- |Enhancement| :class:`SimpleImputer` now warns with feature names
  when features which are skipped due to the lack of any observed
  values in the training set.
  :pr:`21617` by :user:`Christian Ritter <chritter>`.
```
